### DISCUSSION TITLE
Vite imports not recognising .ts files

### DISCUSSION THREAD

**Discussion by maggie44** [quasarframework/quasar#12849] [General - Components / Directives / etc]
Since migrating to Vite, Visual Studio does not seem to recognise the location of .ts files. A clean install with i18n, and in the boot/i18n.ts file I see the following error:

`Cannot find module 'src/i18n' or its corresponding type declarations.ts(2307)`

Same issue occurs on other files too. .vue files are all recognised and imported ok, but .ts files are not. 

Also doesn't recognise JSON files:

`Cannot find module './it/index.json'. Consider using '--resolveJsonModule' to import module with '.json' extension.`

**Comment by maggie44** [comment]
It seems that putting the content from tsconfig.node.ts directly in to tsconfig.ts rather than having tsconfig reference it resolves the issue. Not sure if this has other implications, but it removes the errors at least.

**Comment by lizhongyue248** [comment]
I solved this error by modifying the `tsconfig.node.json` file
```
{
  "compilerOptions": {
    "composite": true,
    "module": "esnext",
    "moduleResolution": "node"
  },
  "include": ["vite.config.ts"]   // Create vite config file and add this.
}

```

**Reply by maggie44** [reply]
And what is in Vite.config.ts?

**Reply by lizhongyue248** [reply]
```
import { defineConfig } from 'vitest/config'
import vue from '@vitejs/plugin-vue'

// https://vitejs.dev/config/
export default defineConfig({
  plugins: [
    vue()
  ]
})

```
